### PR TITLE
Add Provisioning Operation Id to DB Status

### PR DESCRIPTION
**Pull request by shenoypritika**
**What this PR does / why we need it**:
- This PR adds provisioning operation ID field to DB status. This will help us track the operation status for provisioning operations in the upcoming PRs. 

**How Has This Been Tested?**:

image

### CODE REVIEW

**Review by mazin-s — APPROVED** [review]

File: api/v1alpha1/database_types.go [hunk 1]
```diff
@@ -33,11 +33,12 @@ type DatabaseSpec struct {
...
-	Id               string `json:"id"`
-	Status           string `json:"status"`
-	DatabaseServerId string `json:"dbServerId"`
-	Type             string `json:"type"`
+	IPAddress               string `json:"ipAddress"`
+	Id                      string `json:"id"`
+	Status                  string `json:"status"`
+	DatabaseServerId        string `json:"dbServerId"`
+	Type                    string `json:"type"`
+	ProvisioningOperationId string `json:"ProvisioningOperationId"`
```
Comment (manavrajvanshi) [inline api/v1alpha1/database_types.go:41]: Can you please change this to camelCase?

File: config/crd/bases/ndb.nutanix.com_databases.yaml [hunk 2]
```diff
@@ -191,6 +193,7 @@ spec:
               type:
                 type: string
             required:
+            - ProvisioningOperationId
```
Comment (manavrajvanshi) [inline config/crd/bases/ndb.nutanix.com_databases.yaml:196]: Please change the json case to use camelCase and re-run `make generate//manifests` targets to update these values.

**Review by manavrajvanshi — CHANGES_REQUESTED** [review]
Looks good, thanks! 
Please fix the case of the new status field to match json camelCase.

File: api/v1alpha1/database_types.go (hunk 1 above)
Comment (shenoypritika) [reply api/v1alpha1/database_types.go]: Done!

**Review by manavrajvanshi — APPROVED** [review]
LGTM, Thank you!

**Review by yash-ntnx — APPROVED** [review]

**Comment by yash-ntnx** [comment]
changes look good. just want to check if there is something we should UT here. may not be any new test case but augment any of the existing cases. if not, thats ok.

**Pull request merged**